### ISSUE TITLE
Nothing happening on Chrome+Firefox (Angular 2 Final)

### ISSUE THREAD

**Issue by ciesielskico** [Nolanus/ng2-page-scroll#45]
I have followed the setup guide from your README:

1 I have imported ng2pagescroll directly in the module I want to use it.

```
...
import { Ng2PageScrollModule } from 'ng2-page-scroll';

@NgModule({
    imports: [
        SharedModule,
        Ng2PageScrollModule,
        routing
    ],
    exports: [],
    declarations: [
        ReportsComponent,
        ReportListComponent,
        ReportDetailComponent,
        ReportDetailEmptyComponent,
        ChartReportComponent,
        TableReportComponent
    ],
    providers: [
        ReportsService
    ],
})
export class ReportsModule { }
```

2 Added this to my template:

```
<a pageScroll href="#tbl2">
    Tabela 2
</a>

... [lots of tables, so there is a long scrollbar] ...

<h2 id="tbl2">
    tbl2
</h2>
```

but after clicking on the anchor, it does not scroll down. Actually nothing happends, no errors.
Any hints?

PS:
- Adding to AppModule instead of ReportsModule makes the app go to the base URL .
- ReportsModule is lazy-loaded. This should not be a problem though.

**Comment by ciesielskico** [comment]
When I remove the `#` in front of `tbl2` I get following message in console:

`Scrolling not possible, as we can't get any closer to the destination`

But I am at the top of the document, and the `tbl2` id is at the bottom (a couple of view-heights away).

**Comment by Nolanus** [comment]
Hi @ciesielskico,
removing the `#` leads to the target element to scroll to not being found and the console log you just stated.

I quickly took the angular2 guide example for nested modules and created the following plunkr: 
https://plnkr.co/edit/64xjeJAop7dweIxDrQ8S?p=preview
So scrolling inside works, however I haven't tested lazy loaded modules.

Could you please provide some more details about your html layout? Maybe adjusting the "scrollTop" property on the default scrollTarget (body and window elements) does not work for your setup, as the scrollable content is another DOM element?

**Comment by kiuka** [comment]
Hi,

I have the same problem but I tried to scroll via the service.

I am using https://github.com/mgechev/angular-seed with the following code:

`          let pageScrollInstance: PageScrollInstance = PageScrollInstance.advancedInstance(
            this.document,
            '#' + elementId,
            null,
            '',
            null,
            true,
            null,
            500,
            this._scrollEventEmitter
          );

          this.pageScrollService.start(pageScrollInstance);
`

I also experienced that PHPStorm cannot recognize Ng2PageScrollModule when I import it. It says "cannot resolve symbol 'Ng2PageScrollModule'"

**Comment by ciesielskico** [comment]
@Nolanus you are right, my main window  to the app is set to 100vw and 100vh, and inside there are two smaller divs that independently scroll. 

The only solution to my problem I found is to call `.scrollIntoView()`, which is super ugly because it instantly jumps without any animation. But it is the only solution I found after hours of trying out different ways.

**Comment by kiuka** [comment]
In my case the problem was the value of pageScrollInterruptible. I set it false and now everything seems ok.

Is there any way to disable scroll functions till jumping to the section? It is really annoying that I stuck the animation by scrolling the opposite the site.

**Comment by Nolanus** [comment]
@ciesielskico, in case the body/html is not the scrollable element, you may use the [`PageScrollService`](https://github.com/Nolanus/ng2-page-scroll#service) and define your own scroll parents (the element whose `scrollTop` property will be manipulated).

**Comment by Nolanus** [comment]
@kiuka, you may set the duration to 0 to get instant jumping to target instead of animated scrolling.

**Comment by kiuka** [comment]
@Nolanus no, I wanted the animation, therefore I disabled scrolling on mouse wheel event and it solved the problem. I think disable wheel while scrolling would be a very nice feature of this plugin:)

**Comment by Nolanus** [comment]
@kiuka, feel free to create a new issue and describe the feature in more detail!

**Comment by aacercone** [comment]
I am having the same problem with Chrome and Firefox.  Previously it was working until the Final release.  Has there been any progress with this issue?

**Comment by Nolanus** [comment]
Hi @aacercone, 
please make sure that you try to scroll the correct element which is scroll-capable (https://github.com/Nolanus/ng2-page-scroll#using-the-right-scrolling-view). If that's the case the best way to tell whether it is a general library problem or not is to [test the demo application](http://nolanus.github.io/ng2-page-scroll/) or a custom very basic setup.
In case the problem still occurs please provide further details, as I unfortunately did not manage to reproduce the problem (macOS and Windows mixed with Chrome, Firefox, IE/Safari).

**Comment by inzerceubytovani** [comment]
I noticed that Ng2PageScrollModule will stop to work when I set overflow-x: hidden to html,body element. Maybe this information helps somebody.

**Comment by szykov** [comment]
I was facing the same issue like 'nothing happens' and the reason was because you need to import ng2-page-scroll not in an 'AppModule' but in a module where you're going to use it. If you need to use it across many modules then you need to create a shared module and import it any module where you use it.

**Comment by brenikindhi** [comment]
We are facing similar issue, scroll is not working in some windows machines in chrome but works fine in IE. 

Can anyone give inputs on this.

**Comment by samarism** [comment]
I had the same issue and after some debugging found the following: 

In the function `PageScrollInstance.prototype.setScrollPosition` the `scrollTop` value is not getting set for the `scrollingViews` (which are by default `html` & `body`). 

On further probing, I found that the scrollTop was not getting set for body/html because the clientHeight for both were smaller than the target scroll position. When I forcibly set a large height for the body element then it started to scroll.

But, its strange that angular 2 does not update the body's height after rendering all the components.

Hope this helps someone!

**Comment by rmccleave559** [comment]
I had this issue with Chrome, the library worked with Firfeox and IE but this worked for me in Chrome. Not sure if this fix will work for everyone but it managed to get this library working for me, I found that even the window.scrollTo(x,y) method would work in any of my angular components. After playing around with my project I found out it was down to a CSS style that allowed it to scroll. 

position: absolute

So I wrapped the root component of my angular app in a class that provided that position and it works fine now.
